trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/library/AtomicLibraryModel.java

```diff
@@ -456,7 +450,7 @@ private ExpressionResult applyMemoryOrders(final ILocation loc, final Expression
 		// create condition checking whether all memory orders are supported
 		final CPrimitive intType = new CPrimitive(CPrimitives.INT);
 		final Expression seqCst = mExpressionTranslation.constructLiteralForIntegerType(loc, intType,
-				BigInteger.valueOf(MEMORY_ORDER_SEQ_CST));
+				BigInteger.valueOf(Arrays.asList(MEMORY_ORDER).indexOf("memory_order_seq_cst")));
```

Comment:
A Java enum would be more suitable for representation here to access the numbered entries here. With a string array, this can quickly become dangerous if the order of the strings in the array is accidentally changed.

Reply:
But the same could happen, if you accidentally change the order in the enum, right?

Reply:
I would also propose an enum, but additionally, avoid using `indexOf` to get the literal. The enum constructor could take a string argument for the name, and an integer for the literal.

Thinking long-term, we may also want to avoid using literals at some point, and just use symbolic Boogie constants instead. This would be more general, and it would avoid proofs in which e.g. a mysterious `5` appears, when really what we mean is `memory_order_seq_cst`.

Reply:
> But the same could happen, if you accidentally change the order in the enum, right?

Yes, that could happen there too. But it is less likely that a tool like a code formatter changes the order of the elements (since they are numbered implicitly).

Reply:
> I would also propose an enum, but additionally, avoid using `indexOf` to get the literal. The enum constructor could take a string argument for the name, and an integer for the literal. Then we can have getters on the enum to retrieve them.

Okay, I chnaged it as suggested. Retrieving the field names in the correct order requires more work now, therefore `getTypeModels()` is more complicated now -- thus I am still not quite convinced that this is a better solution 😉